yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/ParserImpl.scala

```diff
@@ -132,22 +141,30 @@ final class ParserImpl private (in: Tokenizer) extends Parser:
       Right(Event(EventKind.DocumentStart(explicit = false), token.range))
 
   private def parseDocumentStartOpt(token: Token) = token.kind match
+    case TokenKind.TagDirective(handle, prefix) =>
+      directives.update(handle.value, prefix.value)
+      in.popToken()
+      productions.prepend(ParseDocumentStartOpt) // call self once again
+      getNextEventImpl()
     case TokenKind.DocumentStart =>
       productions.prependAll(ParseDocumentStart :: ParseDocumentStartOpt :: Nil)
       getNextEventImpl()
     case TokenKind.MappingStart | TokenKind.Scalar(_, _) | TokenKind.SequenceStart |
-        TokenKind.FlowMappingStart | TokenKind.FlowSequenceStart | _: TokenKind.Anchor =>
+        TokenKind.FlowMappingStart | TokenKind.FlowSequenceStart | _: TokenKind.Anchor |
+        _: TokenKind.Tag =>
```

Comment:
wdyt about introduce more descriptive type instead of writing `TokenKind.FlowMappingStart TokenKind.FlowSequenceStart | _: TokenKind.Anchor | _: TokenKind.Tag`, but we can do it in another PR.